docs/user_guide/examples.md

```diff
@@ -1,14 +1,248 @@
+
+```python
+import law
+
+from columnflow.tasks.framework.base import AnalysisTask
+
+class MyPrintingTask(AnalysisTask, law.tasks.RunOnceTask):
+
+    @law.tasks.RunOnceTask.complete_on_success
+    def run(self):
+        # Do something here
+        if fail:
+            raise Exception("this task just failed")
+
+    # Alternative
+    def run(self):
+        # Do something here
+        self.mark_complete()
+```
+
+### Add the task to the analysis
+
+After creating a new task, make sure to add the new task to the law config file in your analysis folder, in order to run the task.
+The new task has to be specified under the ```[outputs]``` section as follows:
+
+```ini
+[outputs]
+
+lfn_sources: wlcg_fs_desy_store
+
+cf.MyNewTask: wlcg
+...
+```
+### Executing the Task
+
+after sourcing the law environment, the task can be executed using the command:
+```bash
+law run cf.MyTask --param1 foo
+```
+The parameters can be given using the syntax ```--param1 foo```, where ```param1``` is the name of the parameter and ```foo``` is the value of the parameter.
+If the parameter's name include an underscore, the underscore is replaced by a hyphen (```--param_1 foo``` -> ```--param-1 foo```).
+If the parameter is a boolean, the parameter can be set to ```True``` by only typing ```--param-boolean```.
+If the parameter is a list, the values can be given as a comma-separated list (```--param2 foo,bar,baz```).
 
 ## Custom law.cfg file for users without grid certificate (CMS specific?)
 
-TODO
+Users, who do not have a grid certificate, are not able to run jobs on the grid or store and fetch data from it.
+However, they can still run jobs locally.
+For this purpose, a custom law config file should be created in the analysis folder, which specifies a local dcache and the path to the nanoAoDs for every required campaing, and set the output locations for the tasks to local.
+It is recommended to create a copy of the default law config file and modify it as needed and then redirect the setup to the custom config file by setting the ```LAW_CONFIG_FILE``` environment variable in the `setup.sh` file to the path of the custom config file as follows:
+
+```bash
+export LAW_CONFIG_FILE="${LAW_CONFIG_FILE:-${ANALYSIS_DIR}/law.nocert.cfg}"
+```
+
+The costum config file should contain the following sections:
+```ini
+[core]
+# inherit from the analysis configuration file
+inherit: $HBT_BASE/law.cfg
+
+[local_dcache]
+# define path to the local dcache used later for the lfn (Local File Name) sources 
+base: path/to/local/dcache
+
+[local_campaigns_name]
+base: path/to/local/campaigns/nanoAOD/directory
+
+[wlcg_campaigns_name]
+base: path/to/local/campaigns/nanoAOD/directory
+
+# define additional settings related to caching when working with this campaign 
+use_cache: $CF_WLCG_USE_CACHE
+cache_root: $CF_WLCG_CACHE_ROOT
+cache_cleanup: $CF_WLCG_CACHE_CLEANUP
+cache_max_size: 15GB
+cache_global_lock: True
+cache_mtime_patience: -1
```

Comment:
I wouldn't write about `dcache` since this is also very (DESY) specific. Maybe just talk about "defining different file systems"?